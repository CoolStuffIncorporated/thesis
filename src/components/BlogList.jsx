import React from "react";

const BlogList = blogs => (
  <ul>
    {blogs.map(({ id, title, author }) => (
      <li key={id}>
        {title}-{author}
      </li>
    ))}
  </ul>
);

export default BlogList;
