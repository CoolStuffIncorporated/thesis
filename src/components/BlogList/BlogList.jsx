import React from "react";
import PropTypes from "prop-types";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import { withTheme } from "@material-ui/core/styles";

const styles = {
  marginLeft: "130px",
  marginRight: "130px"
};

// can eventually add blog media, user avatar, etc.
const BlogList = ({ blogs }) => {
  return blogs.blogs.map(blog => (
    <Card key={blog.blog_id} style={styles}>
      <CardHeader title={blog.title} />
      <CardContent>
        <Typography paragraph>{blog.content}</Typography>
      </CardContent>
    </Card>
  ));
};

// BlogList.propTypes = {
//   blogs: PropTypes.object
// };

// BlogList.defaultProps = {
//   blogs: [
//     {
//       blog_id: 0,
//       title: "Follow some blogs!",
//       blogAuthor: "",
//       content: "Nothing here yet"
//     }
//   ]
// };

export default withTheme()(BlogList);
