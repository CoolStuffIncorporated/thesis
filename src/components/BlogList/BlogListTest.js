/* global expect, it, describe, jest */
import React from 'react';
import { shallow, mount } from 'enzyme';
import BlogList from './BlogList';

describe('BlogList component', () => {
  it('Should render successfully', () => {
    const component = shallow(<BlogList />);
    expect(component.exists()).toEqual(true);
  });
});
