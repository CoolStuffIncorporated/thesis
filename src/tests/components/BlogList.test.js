/* global expect, test, shallow */
import React from "react";
import BlogList from "../../components/BlogList/BlogList";

const testBlogs = [
  {
    blogId: "1",
    blogTitle: "WELCOME TO NIHON",
    blogAuthor: "BROICHI",
    blogContents: "YOLO SWAG"
  }
];
test("BlogList renders without crashing", () => {
  const component = shallow(<BlogList />);
  expect(component.exists()).toEqual(true);
});
test("BlogList renders a blog post", () => {
  const component = shallow(
    <BlogList blogs={testBlogs} />
  );
  expect(component).toMatchSnapshot();
});
