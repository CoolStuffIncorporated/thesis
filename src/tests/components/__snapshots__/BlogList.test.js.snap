// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList renders a blog post 1`] = `
<ul>
  <WithStyles(Card)
    key="1"
  >
    <WithStyles(CardHeader)
      subheader="BROICHI"
      title="WELCOME TO NIHON"
    />
    <WithStyles(CardContent)>
      <WithStyles(Typography)
        paragraph={true}
      >
        YOLO SWAG
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</ul>
`;
